File: sqlparser/src/sql/SqlOperationColumn.java
//@
package cdata.sql;
//@
/*#
using RSSBus.core;
using RSSBus;
namespace CData.Sql {
#*/

public final class SqlOperationColumn extends SqlColumn implements ISqlElement {
  private final SqlOperationExpression expr;

  public SqlOperationColumn(SqlOperationExpression val) {
    super(null, null);
    this.expr = val;
  }

  public SqlOperationColumn(String alias, SqlOperationExpression val) {
    super(alias, alias);
    this.expr = val;
  }

  public /*#override#*/ SqlExpression getExpr() {
    return this.expr;
  }

  public /*#override#*/ boolean equals(String name) {
    return false;
  }

  public /*#override#*/ boolean isEvaluatable() {
    return expr != null && expr.isEvaluatable();
  }

  public /*#override#*/ SqlValue evaluate() {
    if (isEvaluatable()) {
      return expr.evaluate();
    }
    return SqlValue.getNullValueInstance();
  }

  public /*#override#*/ Object clone() {
    SqlOperationColumn obj = new SqlOperationColumn(this.getColumnName(), this.expr);
    obj.copy(this);
    return obj;
  }

  public /*#override#*/ boolean hasAlias() {
    if (getAlias() != null) {
      return true;
    } else {
      return false;
    }
  }
}

File: sqlparser/src/sql/csgen/SqlOperationColumn.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
namespace CData.Sql {


public sealed class SqlOperationColumn : SqlColumn , ISqlElement {
  private readonly SqlOperationExpression expr;

  public SqlOperationColumn(SqlOperationExpression val) : base(null, null) {
    this.expr = val;
  }

  public SqlOperationColumn(string alias, SqlOperationExpression val) : base(alias, alias) {
    this.expr = val;
  }

  public override SqlExpression GetExpr() {
    return this.expr;
  }

  public override bool Equals(string name) {
    return false;
  }

  public override bool IsEvaluatable() {
    return expr != null && expr.IsEvaluatable();
  }

  public override SqlValue Evaluate() {
    if (IsEvaluatable()) {
      return expr.Evaluate();
    }
    return SqlValue.GetNullValueInstance();
  }

  public override Object Clone() {
    SqlOperationColumn obj = new SqlOperationColumn(this.GetColumnName(), this.expr);
    obj.Copy(this);
    return obj;
  }

  public override bool HasAlias() {
    if (GetAlias() != null) {
      return true;
    } else {
      return false;
    }
  }
}
}

